src/main/java/org/example/bookstore/validator/FieldMatchValidator.java

```diff
@@ -0,0 +1,37 @@
+package org.example.bookstore.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+import java.util.Objects;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstFieldName = constraintAnnotation.first();
+        secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Class<?> clazz = value.getClass();
+        try {
+            Field firstFieldValue = clazz.getDeclaredField(firstFieldName);
```

Comment:
that is not a field value yet, just a `Field` object, same for the other one
```suggestion
            Field firstField = clazz.getDeclaredField(firstFieldName);
```